pkg/config/types.go

```diff
@@ -59,8 +65,13 @@ type Config interface {
 	ConfigFileUsed() string
 
 	BindPFlag(key string, flag *pflag.Flag) error
+}
 
-	// Datadog specific
+// DDConfigHelpers describes the config helpers that are not provided by viper.Viper
+type DDConfigHelpers interface {
+	// GetConfigEnvVars returns a list of the non-sensitive env vars that the config supports
 	GetConfigEnvVars() []string
-	bindEnvAndSetDefault(key string, val interface{})
+	// BindEnvAndSetDefault sets the default value for a config parameter and adds an env binding
+	// in one call
+	BindEnvAndSetDefault(key string, val interface{})
```

Comment:
It looks like this just calls `BindEnv` and `SetDefault` internally, so it might be that we could just remove it and instead just add this other method (GetEnvVars) into the original interface. That'd remove a lot of extra complications imho.